crates/stackable-telemetry/src/tracing/settings/file_log.rs

```diff
@@ -0,0 +1,155 @@
+pub struct FileLogSettings {
+    /// Common subscriber settings that apply to the File Log Subscriber.
+    pub common_settings: Settings,
+
+    /// Path to directory for log files.
+    pub file_log_dir: PathBuf,
+}
+
+impl Deref for FileLogSettings {
+    type Target = Settings;
+
+    fn deref(&self) -> &Self::Target {
+        &self.common_settings
+    }
+}
+
+/// This trait is only used for the typestate builder and cannot be implemented
+/// outside of this crate.
+///
+/// The only reason it has pub visibility is because it needs to be at least as
+/// visible as the types that use it.
+#[doc(hidden)]
+pub trait BuilderState: private::Sealed {}
+
+/// This private module holds the [`Sealed`][1] trait that is used by the
+/// [`BuilderState`], so that it cannot be implemented outside of this crate.
+///
+/// We impl Sealed for any types that will use the trait that we want to
+/// restrict impls on. In this case, the [`BuilderState`] trait.
+///
+/// [1]: private::Sealed
+#[doc(hidden)]
+mod private {
+    use super::*;
+
+    pub trait Sealed {}
+
+    impl Sealed for builder_state::PreLogDir {}
+    impl Sealed for builder_state::Config {}
+}
+
+/// This module holds the possible states that the builder is in.
+///
+/// Each state will implement [`BuilderState`] (with no methods), and the
+/// Builder struct ([`FileLogSettingsBuilder`][1]) itself will be implemented with
+/// each state as a generic parameter.
+/// This allows only the methods to be called when the builder is in the
+/// applicable state.
+///
+/// [1]: super::FileLogSettingsBuilder
+#[doc(hidden)]
+pub mod builder_state {
+    /// The initial state, before the log directory path is set.
+    #[derive(Default)]
+    pub struct PreLogDir;
+
+    /// The state that allows you to configure the [`FileLogSettings`][1].
+    ///
+    /// [1]: super::FileLogSettings
+    #[derive(Default)]
+    pub struct Config;
+}
+
+// Make the states usable
+#[doc(hidden)]
+impl BuilderState for builder_state::PreLogDir {}
+
+#[doc(hidden)]
+impl BuilderState for builder_state::Config {}
+
+/// For building [`FileLogSettings`].
+///
+/// <div class="warning">
+/// Do not use directly, instead use the [`Settings::builder`] associated function.
+/// </div>
+pub struct FileLogSettingsBuilder<S: BuilderState> {
+    pub(crate) common_settings: Settings,
+    pub(crate) file_log_dir: Option<PathBuf>,
+
+    /// Allow the generic to be used (needed for impls).
+    _marker: std::marker::PhantomData<S>,
+}
+
+impl FileLogSettingsBuilder<builder_state::PreLogDir> {
+    /// Set the directory for log files.
+    ///
+    /// A directory is required for using the File Log subscriber.
+    pub fn with_file_log_dir(self, path: String) -> FileLogSettingsBuilder<builder_state::Config> {
+        FileLogSettingsBuilder {
+            common_settings: self.common_settings,
+            file_log_dir: Some(PathBuf::from(path)),
+            _marker: std::marker::PhantomData,
+        }
+    }
```

Comment:
suggestion: We should use an actual path here instead of a string.

```suggestion
    pub fn with_file_log_dir<P>(self, path: P) -> FileLogSettingsBuilder<builder_state::Config> where P: AsRef<Path> {
        FileLogSettingsBuilder {
            common_settings: self.common_settings,
            file_log_dir: path.as_ref().into_path_buf(),
            _marker: std::marker::PhantomData,
        }
    }
```